app/auth.py

```diff
@@ -151,13 +153,31 @@ def logout():
     return {"message": "Provide a valid authentication token"}, status.HTTP_403_FORBIDDEN
 
 
-@app.route('/api/auth/reset-password/', methods=['POST'])
+@app.route('/api/auth/reset-password/', methods=['PUT'])
 def reset_password():
     """Reset user Password endpoint takes in a password and resets the password"""
-    if 'user' in session:
-        password = request.data.get('password')
-        Users.user_db['user'] = password
-        message = {"message": "you have succesfuly reset your password"}
-        return message, status.HTTP_200_OK
-    message = {"message": "you need to log in first to reset password"}
-    return message, status.HTTP_401_UNAUTHORIZED
+    password = request.data.get('password')
+    access_token = authentication_request()
+
+    if access_token:
+        # Attempt to decode the token and get the User ID
+        user_id = Users.decode_token(access_token)
+        if not isinstance(user_id, str):
+            user = Users.query.filter_by(id=user_id).first()
+            try:
+                if not user:
+                    raise exeptions.NotFound()
+                
+                user.password = Users.hash_password(password)
```

Comment:
I don't think that you have validated the new password that you are resetting to.

Reply:
all right that escaped me thanks